esda/correlogram.py

```diff
@@ -0,0 +1,144 @@
+# Spatial Correlograms
+
+import geopandas as gpd
+import pandas as pd
+import numpy as np
+from joblib import Parallel, delayed
+from libpysal.cg.kdtree import KDTree
+from libpysal.weights import KNN, DistanceBand
+from libpysal.weights.util import get_points_array
+
+from .geary import Geary
+from .getisord import G
+from .moran import Moran
+
+
+def _get_autocorrelation_stat(inputs):
+    """helper function for computing parallel autocorrelation statistics
+
+    Parameters
+    ----------
+    inputs : tuple
+        tuple of (y, tree, W, statistic, STATISTIC, dist, weights_kwargs, stat_kwargs)
+
+    Returns
+    -------
+    pandas.Series
+        a pandas series with the computed autocorrelation statistic and its simulated p-value
+    """
+    (
+        y,  # y variable
+        tree,  # kd tree
+        W,  # weights class DistanceBand or KNN
+        STATISTIC,  # class of statistic (Moran, Geary, etc)
+        dist,  # threshold/k parameter for the weights
+        weights_kwargs,  # additional args
+        stat_kwargs,  # additional args
+    ) = inputs
+
+    w = W(tree, dist, silence_warnings=True, **weights_kwargs)
+    autocorr = STATISTIC(y, w, **stat_kwargs)
+    attrs = []
+    all_attrs = list(dict(vars(autocorr)).keys())
+    for attribute in all_attrs:
+        attrs.append(getattr(autocorr, str(attribute)))
```

Comment:
I wonder if this hits any attributes that are not scalar values, and what that may do?

Reply:
i went through a few drafts first. It does grab a few, (in particular some of the attributes on the original Moran etc object are arrays, so they end up as lists in each row. not idea). So I filter them out down in line 142 (and also drop a few constants like `n` and `permutations`)